File: backend/app/detection/mitre.py
"""
MITRE ATT&CK technique reference — deliberately NOT the full ~600-technique
STIX bundle. This is a curated lookup for exactly the techniques this
phase's rules reference, so alerts can show a human-readable name
alongside the technique ID without shipping a dataset most of which would
never be used.

Importing the full official ATT&CK STIX bundle (via the attackcti or
mitreattack-python packages) is a legitimate stretch goal once more rule
coverage exists — noted here rather than silently deferred.
"""

MITRE_TECHNIQUES: dict[str, str] = {
    "T1110": "Brute Force",
    "T1110.001": "Brute Force: Password Guessing",
    "T1059": "Command and Scripting Interpreter",
    "T1071": "Application Layer Protocol",
}


def technique_name(technique_id: str) -> str:
    return MITRE_TECHNIQUES.get(technique_id, technique_id)

File: frontend/src/lib/mitre.ts
// Mirrors backend/app/detection/mitre.py's curated lookup exactly — same
// deliberate scope (only techniques this project's rules actually
// reference, not the full ATT&CK bundle). Keep these two files in sync by
// hand for now; see lib/types.ts for the same trade-off note.

export const MITRE_TECHNIQUES: Record<string, string> = {
  T1110: "Brute Force",
  "T1110.001": "Brute Force: Password Guessing",
  T1059: "Command and Scripting Interpreter",
  T1071: "Application Layer Protocol",
};

export function techniqueName(techniqueId: string): string {
  return MITRE_TECHNIQUES[techniqueId] ?? techniqueId;
}
